app/controllers/items_controller.rb

```diff
@@ -0,0 +1,48 @@
+class ItemsController < ApplicationController
+  def index
+    if params[:merchant_id] == nil
+      @items = Item.all
+    else
+      @merchant = Merchant.find(params[:merchant_id])
+      @items = @merchant.items
+    end
+  end
+
+  def show
+    @item = Item.find(params[:id])
+  end
+
+  def new
+    @merchant_id = params[:merchant_id]
+  end
+
+  def create
+    merchant = Merchant.find(params[:merchant_id])
+    item = Item.create!(item_params)
+    item.save!
```

Comment:
Here is a great opportunity for dealing with a sad path. If the item_params had some bad data in it, for instance if the name was blank, then this save would fail and would throw an error because you have the bang (!) on it. If the item does not save correctly, `item.save` will return false, so you can do:

```
if item.save
# Code here for successful item creation
else
# Code here for unsuccessful item creation (sad path) 
end
```